src/subdomains/core/buy-crypto/process/services/buy-crypto-webhook.service.ts

```diff
@@ -0,0 +1,34 @@
+import { Injectable } from '@nestjs/common';
+import { PaymentWebhookState } from 'src/subdomains/generic/user/services/webhook/dto/payment-webhook.dto';
+import { WebhookService } from 'src/subdomains/generic/user/services/webhook/webhook.service';
+import { BuyCrypto } from '../entities/buy-crypto.entity';
+import { CheckStatus } from '../enums/check-status.enum';
+
+@Injectable()
+export class BuyCryptoWebhookService {
+  constructor(private readonly webhookService: WebhookService) {}
+
+  async triggerWebhook(buyCrypto: BuyCrypto, webhookState: PaymentWebhookState): Promise<void> {
+    buyCrypto.isCryptoCryptoTransaction
+      ? this.webhookService.cryptoCryptoUpdate(buyCrypto.user, buyCrypto, webhookState)
+      : this.webhookService.fiatCryptoUpdate(buyCrypto.user, buyCrypto, webhookState);
+  }
+
+  public getWebhookState(buyCrypto: BuyCrypto): PaymentWebhookState {
+    if (buyCrypto.chargebackDate) return PaymentWebhookState.RETURNED;
+
+    switch (buyCrypto.amlCheck) {
+      case CheckStatus.PENDING:
+        return PaymentWebhookState.AML_PENDING;
+      case CheckStatus.FAIL:
+        return PaymentWebhookState.FAILED;
+      case CheckStatus.PASS:
+        if (buyCrypto.isComplete) return PaymentWebhookState.COMPLETED;
+        break;
+    }
+
+    if (buyCrypto.outputReferenceAsset) return PaymentWebhookState.PROCESSING;
+
+    return PaymentWebhookState.CREATED;
+  }
```

Comment:
Maybe we should use this method for all `triggerWebhook` calls and remove the `webhookState: PaymentWebhookState` parameter? It would be easier to understand the webhook types and mappings.